akka-typed-tests/src/test/scala/akka/typed/cluster/sharding/ClusterShardingApiSpec.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+
+package akka.typed.cluster.sharding
+
+import akka.typed.ActorSystem
+import akka.typed.cluster.Cluster
+import akka.typed.scaladsl.Actor
+import akka.typed.scaladsl.adapter._
+
+class ClusterShardingApiSpec {
+
+  // Compile only for now
+
+  val system: akka.actor.ActorSystem = ???
+  val typedSystem: ActorSystem[Nothing] = system.toTyped
+  val cluster = Cluster(typedSystem)
+
+  trait EntityProtocol
+
+  case class Add(thing: String) extends EntityProtocol
+
+  case object PassHence extends EntityProtocol
+
+  val entityBehavior =
+    Actor.deferred[EntityProtocol] { _ ⇒
+      var things: List[String] = Nil
```

Comment:
we should write tests in good style, this is a typical `immutable` and no need for `deferred`, unless there is a reason for testing with `deferred`

Reply:
Nope no reason for it I think, this was copied from @johanandren's initial API design spec, so I guess it's not binding, mostly a playground -- I'll delete this once I have real tests